bolt-design-system/bolt
# Table component

## Jira

http://vjira2:8080/browse/BDS-529

## Summary

Created a new table component that can handle simple or extremely complex tabula data.

## Details

1. `bolt-table` is added to components.
2. There are two opinionated formats: regular and numeric. Numeric is great for number focused tables. Regular is suited for more text heavy tables. Format prop is a string for scalability, there could be more formats down the road.
3. There is a borderless version that removes the border in between cells.
4. There is a prop to make the first column fit to the widest text on that column. Note that side headers do not count as a column.
5. Attributes are supported on table cells, so `rowspan` and `colspan` are fully functional.
6. Semantic markup is supported, tables can have top headers, side headers, and footer.

## How to test

1. Pull down the branch and navigate to the table view all page, inspect all the styles demonstrated there.
2. Check the twig file for logics and conditions.
3. Check for standards and conventions.
